terraform/table_terraform_output.go

```diff
@@ -87,42 +89,72 @@ type terraformOutput struct {
-	// The path comes from a parent hydate, defaulting to the config paths or
+	// The path comes from a parent hydrate, defaulting to the config paths or
 	// available by the optional key column
-	data := h.Item.(filePath)
-	path := data.Path
+	pathInfo := h.Item.(filePath)
+	path := pathInfo.Path
 
 	// Return if the path is a TF plan path
-	if data.IsTFPlanFilePath {
+	if pathInfo.IsTFPlanFilePath {
 		return nil, nil
 	}
 
-	combinedParser, err := Parser()
-	if err != nil {
-		plugin.Logger(ctx).Error("terraform_output.listOutputs", "create_parser_error", err)
-		return nil, err
-	}
-
 	content, err := os.ReadFile(path)
 	if err != nil {
 		plugin.Logger(ctx).Error("terraform_output.listOutputs", "read_file_error", err, "path", path)
 		return nil, err
 	}
 
-	var tfOutput terraformOutput
+	var docs []model.Document
+
+	// Check if the file contains TF state
+	if pathInfo.IsTFStateFilePath {
+		// Initialize the JSON parser
+		jsonParser := p.Parser{}
 
-	for _, parser := range combinedParser {
-		parsedDocs, err := ParseContent(ctx, d, path, content, parser)
+		// Parse the file content using the JSON parser
+		var str string
+		documents, _, err := jsonParser.Parse(str, content)
 		if err != nil {
-			plugin.Logger(ctx).Error("terraform_output.listOutputs", "parse_error", err, "path", path)
-			return nil, fmt.Errorf("failed to parse file %s: %v", path, err)
+			plugin.Logger(ctx).Error("terraform_output.listOutputs", "state_parse_error", err, "path", path)
+			return nil, fmt.Errorf("failed to parse state file %s: %v", path, err)
+		}
+
+		docs = append(docs, documents...)
+	} else {
+		// Build the terraform parser
+		combinedParser, err := Parser()
+		if err != nil {
+			plugin.Logger(ctx).Error("terraform_output.listOutputs", "create_parser_error", err)
+			return nil, err
+		}
+
+		for _, parser := range combinedParser {
+			parsedDocs, err := ParseContent(ctx, d, path, content, parser)
+			if err != nil {
+				plugin.Logger(ctx).Error("terraform_output.listOutputs", "parse_error", err, "path", path)
+				return nil, fmt.Errorf("failed to parse file %s: %v", path, err)
+			}
+			docs = append(docs, parsedDocs.Docs...)
 		}
+	}
 
-		for _, doc := range parsedDocs.Docs {
-			if doc["output"] != nil {
-				// For each output, scan its arguments
-				for outputName, outputData := range doc["output"].(model.Document) {
-					tfOutput, err = buildOutput(ctx, path, content, outputName, outputData.(model.Document))
+	for _, doc := range docs {
+		if doc["output"] != nil {
+			// For each output, scan its arguments
+			for outputName, outputData := range doc["output"].(model.Document) {
+				tfOutput, err := buildOutput(ctx, pathInfo.IsTFStateFilePath, path, content, outputName, outputData.(model.Document))
+				if err != nil {
+					plugin.Logger(ctx).Error("terraform_output.listOutputs", "build_output_error", err)
+					return nil, err
+				}
+				d.StreamListItem(ctx, tfOutput)
+			}
+		} else if doc["outputs"] != nil {
```

Comment:
Could TF configuration files have `outputs` too? And when would a state file have `output` vs. `outputs`?

Reply:
TF configuration files can not have outputs. State files are always generated with outputs.